twoliter/src/cmd/build.rs

```diff
@@ -23,12 +37,102 @@ pub(crate) struct BuildVariant {
     project_path: Option<PathBuf>,
 
     /// The architecture to build for.
-    #[clap(long = "arch", default_value = "x86_64")]
+    #[clap(long = "arch", env = "BUILDSYS_ARCH", default_value = "x86_64")]
     arch: String,
+
+    /// The variant to build.
+    #[clap(env = "BUILDSYS_VARIANT")]
+    variant: String,
+
+    /// The go modules that should be build
+    #[clap(long = "go-modules", env = "GO_MODULES", default_value = "")]
+    go_modules: String,
```

Comment:
I'd rather not expose this here. Potentially we want a contract like:
> Twoliter will search the `sources` path for `Cargo.lock` and `go.mod` files and download the referenced Rust crates and Go modules prior to package builds.

And eventually support for `.twoliterignore` files to suppress this behavior by excluding certain paths.

Reply:
> I'd rather not expose [Go Modules] here. Potentially we want a contract like:

This is tricky because it either has to be implemented now, or there has to be a way to provide the list of Go modules (because the build system currently operates using a list of Go modules, effectively hard-coded in the Makefile, but which can be overridden).

A possible middle ground would be to include it here with a high priority follow-up issue to remove it.

Reply:
Oh look, I have a very well-written (:sweat_smile: ) issue for this already. https://github.com/bottlerocket-os/twoliter/issues/88#issue-1918319719